### ISSUE TITLE
Code generator cannot handle tables from other schemas than PUBLIC referencing H2 enum domain types

### ISSUE THREAD

**Issue by muued** [jOOQ/jOOQ#11722]
### Expected behavior
The field type of enum domains should be correctly identified even when used from a schema other than public.
I would have expected the use of DataType::asEnumDataType at some point.

### Actual behavior
The field type is determined via a reference to the domain as `PUBLIC.domainname`.
Prefixing it with PUBLIC. seems to be illegal in H2.
DataType::asEnumDataType is not involved in the process.

### Steps to reproduce the problem

- If the problem relates to code generation, please post your code generation configuration
see MCVE
it includes a forcedType entry for the enum on the `problemcolumn`
this is not active for the column `notbetter`

- A complete set of DDL statements can help re-create the setup you're having
see MCVE, main aspect is creating a table in a schema other than public and using an enum domain as column type
```
CREATE DOMAIN enumdomain AS ENUM ('x', 'y');
CREATE SCHEMA mcve;
CREATE TABLE mcve.test ( problemcolumn enumdomain );
```

- An MCVE can be helpful to provide a complete reproduction case
https://github.com/muued/jOOQ-mcve/tree/issue11722

### Versions

- jOOQ: 3.14.8
- Java: 15
- Database (include vendor): H2 1.4.200
- OS: macOS 10.15.7
- JDBC Driver (include name if inofficial driver):

**Comment by lukaseder** [comment]
Thanks a lot for your report. I'm not sure if we can fix this on our side, see these issues for a bit of history: 

- https://github.com/h2database/h2database/issues/2809
- https://github.com/h2database/h2database/issues/1507

A future H2 version will support qualified domain types, in case of which our implementation will be correct. We could, of course, try to adapt the current support to the current H2 version, having to fix it again later when the next H2 version is published, or support multiple H2 versions in parallel. But that sounds like too much busywork for this particular set of features.

I'll keep this issue open to make sure we verify this works correctly once it should work.

**Comment by lukaseder** [comment]
H2 2.0.202 has now been released and it cleaned up quite a bit on how it represents `ENUM` and `DOMAIN` types in the `INFORMATION_SCHEMA`. Once we support that H2 version in jOOQ 3.16, there's probably room for fixing this issue here.

See also: https://github.com/jOOQ/jOOQ/issues/9609

**Comment by lukaseder** [comment]
Still doesn't work with the H2 2.0.202 support yet. Something is wrong in the generated code. Will investigate soon

**Comment by lukaseder** [comment]
Fixed in jOOQ 3.17.0 and 3.16.5 (https://github.com/jOOQ/jOOQ/issues/13042)
